packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,324 @@
+            case keyCodeHome:
+                event.preventDefault();
+                this.activeTabIndex = 0;
+                this.setTabs();
+                break;
+            case keyCodeEnd:
+                event.preventDefault();
+                this.activeTabIndex = tabsLength - 1;
+                this.setTabs();
+                break;
+        }
+    };
+
+    private handleActiveIndicatorPosition() {
+        if (this.activeIndicator) {
+            if (this.ticking) {
+                this.activeIndicatorRef.style.transform = "translateX(0px)";
+                this.activeIndicatorRef.classList.remove("activeIndicatorTransition");
+                if (this.isHorizontal()) {
+                    this.activeIndicatorRef.style.gridColumn = `${
+                        this.activeTabIndex + 1
+                    }`;
+                } else {
+                    this.activeIndicatorRef.style.gridRow = `${this.activeTabIndex + 1}`;
+                }
+                this.ticking = false;
+            } else {
+                this.ticking = true;
+
+                this.animateActiveIndicator();
+            }
+        }
+    }
+
+    private animateActiveIndicator(): void {
+        if (this.isHorizontal()) {
+            const prev: number = this.activeIndicatorRef.offsetLeft;
+            this.activeIndicatorRef.style.gridColumn = `${this.activeTabIndex + 1}`;
+            const next: number = this.activeIndicatorRef.offsetLeft;
+            this.activeIndicatorRef.style.gridColumn = `${this.prevActiveTabIndex + 1}`;
+            const dif: number = next - prev;
+            this.activeIndicatorRef.style.transform = `translateX(${dif}px)`;
+            this.activeIndicatorRef.classList.add("activeIndicatorTransition");
+            this.activeIndicatorRef.addEventListener("transitionend", () => {
+                this.ticking = false;
+                this.activeIndicatorRef.style.gridColumn = `${this.activeTabIndex + 1}`;
+                this.activeIndicatorRef.style.transform = "translateX(0px)";
+                this.activeIndicatorRef.classList.remove("activeIndicatorTransition");
+            });
+        } else {
+            const prev: number = this.activeIndicatorRef.offsetTop;
+            this.activeIndicatorRef.style.gridRow = `${this.activeTabIndex + 1}`;
+            const next: number = this.activeIndicatorRef.offsetTop;
+            this.activeIndicatorRef.style.gridRow = `${this.prevActiveTabIndex + 1}`;
+            const dif: number = next - prev;
+            this.activeIndicatorRef.style.transform = `translateY(${dif}px)`;
+            this.activeIndicatorRef.classList.add("activeIndicatorTransition");
+            this.activeIndicatorRef.addEventListener("transitionend", () => {
+                this.ticking = false;
+                this.activeIndicatorRef.style.gridRow = `${this.activeTabIndex + 1}`;
+                this.activeIndicatorRef.style.transform = "translateX(0px)";
```

Comment:
should this be `translateY`?

Reply:
Its `translateX` if `orientation` is set to `horizontal`, `translateY` is orientation is set to `vertical`